src/pathfind/teleport.cpp

```diff
@@ -215,56 +215,41 @@ teleport_map::teleport_map(
 			}
 		}
 
-		std::string teleport_id = group.get_teleport_id();
 		std::set<map_location>::iterator source_it = locations.first.begin();
 		for (; source_it != locations.first.end(); ++source_it ) {
-			if(teleport_map_.count(*source_it) == 0) {
-				std::set<std::string> id_set;
-				id_set.insert(teleport_id);
-				teleport_map_.emplace(*source_it, id_set);
+			auto map_it = teleport_map_.find(*source_it);
+
+			if(map_it == teleport_map_.end()) {
+				std::unordered_set<map_location> map_locations;
+				map_locations.insert(locations.second.begin(), locations.second.end());
+				teleport_map_.emplace(*source_it, map_locations);
```

Comment:
If you want the local object, it should be `std::move`-ed into `teleport_map_`. If not, suggest:
```cpp
teleport_map_.emplace(*source_it, { locations.second.begin(), locations.second.end() });
```

Reply:
The suggested code didnt work for me: error: no matching function for call to ‘std::unordered_map<map_location, std::unordered_set<map_location> >::emplace(const map_location&, <brace-enclosed initializer list>)’

Reply:
Is 'teleport_map_.emplace(*source_it, std::unordered_set<map_location>(locations.second.begin(), locations.second.end()));' good or dies it have to be 'teleport_map_.emplace(*source_it, std::move(std::unordered_set<map_location>(locations.second.begin(), locations.second.end())));'? Also doesnt the compiler optimize these situations so there wouldnt be any copying when it can be avoided?

Reply:
> teleport_map_.emplace(*source_it, std::unordered_set<map_location>(locations.second.begin(), locations.second.end()));

This should be fine, but also check if it works when you delete `<map_location>`.

Reply:
```cpp
teleport_map_.emplace(*source_it, std::unordered_set(locations.second.begin(), locations.second.end()));
```